internal/pkg/service/common/etcdop/watchmirror_test.go

```diff
@@ -102,3 +105,80 @@ func TestMirror(t *testing.T) {
 		"Luke Blue":   30,
 	}, mirror.ToMap())
 }
+
+func TestFullMirror(t *testing.T) {
+	t.Parallel()
+
+	wg := &sync.WaitGroup{}
+	defer wg.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+
+	// Create a typed prefix with some keys
+	client := etcdhelper.ClientForTest(t, etcdhelper.TmpNamespace(t))
+	pfx := NewTypedPrefix[testUser]("my/prefix", serde.NewJSON(serde.NoValidation))
+	require.NoError(t, pfx.Key("key1").Put(client, testUser{FirstName: "John", LastName: "Brown", Age: 10}).Do(ctx).Err())
+	require.NoError(t, pfx.Key("key2").Put(client, testUser{FirstName: "Paul", LastName: "Green", Age: 20}).Do(ctx).Err())
+
+	// Setup full mirroring of the etcd prefix tree to the memory.
+	logger := log.NewDebugLogger()
+	mirror, errCh := SetupFullMirror(logger, pfx.GetAllAndWatch(ctx, client, clientv3.WithPrevKV())).
+		WithFilter(func(event WatchEventT[testUser]) bool {
+			return !strings.Contains(event.Kv.String(), "/ignore")
+		}).
+		StartMirroring(ctx, wg)
```

Comment:
```suggestion
	mirror, errCh := SetupFullMirror(
	        logger, 
	        pfx.GetAllAndWatch(ctx, client, clientv3.WithPrevKV())).
		WithFilter(func(event WatchEventT[testUser]) bool {
			return !strings.Contains(event.Kv.String(), "/ignore")
		}).
		StartMirroring(ctx, wg)
```
For better comparison between tests

Reply:
:heavy_check_mark: